tests/test_network.py

```diff
@@ -1213,3 +1213,82 @@ def test_clone_network_with_scoped_attributes_to_new_project(self, client, netwo
         #this project (as it was created outside the project test suite) does not have a default scoped
         #attributes like the projectes created using the ProjectMaker, hence it will have 0
         assert len(project_scoped_attributes) == 0
+
+
+    def test_clone_node(self, client, network_with_data):
+
+        node_to_clone = network_with_data.nodes[0]
+
+        cloned_node_id = client.clone_node(node_to_clone.id)
+
+        cloned_node = client.get_node(cloned_node_id)
+
+        assert cloned_node.name == f"{node_to_clone.name} (1)"
+
+        assert len(cloned_node.attributes) == len(node_to_clone.attributes)
+
+        scenario = client.get_scenario(network_with_data.scenarios[0].id)
+        original_node_data = list(filter(lambda x: x.resource_attr_id in [a.id for a in node_to_clone.attributes],
+                                         scenario.resourcescenarios))
+
+        cloned_node_data = list(filter(lambda x: x.resource_attr_id in [a.id for a in cloned_node.attributes],
+                                       scenario.resourcescenarios))
+
+        assert len(cloned_node_data) == len(original_node_data)-1 #has no outputs, so has one less dataset
+
+
+        cloned_node_id_2 = client.clone_node(node_to_clone.id, include_outputs=True)
+
+        cloned_node_2 = client.get_node(cloned_node_id_2)
+
+        assert cloned_node_2.name == f"{node_to_clone.name} (2)"
+
+        assert len(cloned_node_2.attributes) == len(node_to_clone.attributes)
+
+        scenario = client.get_scenario(network_with_data.scenarios[0].id)
+        original_node_data = list(filter(lambda x: x.resource_attr_id in [a.id for a in node_to_clone.attributes],
+                                         scenario.resourcescenarios))
+
+        cloned_node_data = list(filter(lambda x: x.resource_attr_id in [a.id for a in cloned_node_2.attributes],
+                                       scenario.resourcescenarios))
+
+        assert len(cloned_node_data) == len(original_node_data) #has no outputs, so has one less dataset
```

Comment:
```suggestion
        assert len(cloned_node_data) == len(original_node_data)
```

Outputs now included as per kwarg so lengths should be the same.